nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/trash/test/TestTrashBackwardMechanism.java

```diff
@@ -62,19 +63,55 @@ public void testPropertyTrashService() {
 
     @Test
     public void testMechanism() {
-        DocumentModel doc = session.createDocumentModel("/", "file001", "File");
-        doc = session.createDocument(doc);
+        DocumentModel folder = session.createDocumentModel("/", "folder", "Folder");
+        DocumentModel file1 = session.createDocumentModel("/folder", "file1", "File");
+        DocumentModel file2 = session.createDocumentModel("/folder", "file2", "File");
+        folder = session.createDocument(folder);
+        file1 = session.createDocument(file1);
+        file2 = session.createDocument(file2);
         session.save();
 
         // follow delete transition to check if TrashService is called
-        session.followTransition(doc, LifeCycleConstants.DELETE_TRANSITION);
+        TrashService trashService = Framework.getService(TrashService.class);
+        trashService.trashDocument(file1);
 
-        coreFeature.waitForAsyncCompletion();
+        txFeature.nextTransaction();
 
         // we assert ecm:isTrashed property in order to confirm that TrashService has been called
-        assertTrue(session.isTrashed(doc.getRef()));
-        Boolean isTrashed = session.getDocumentSystemProp(doc.getRef(), SYSPROP_IS_TRASHED, Boolean.class);
+        file1 = session.getDocument(file1.getRef());
+        assertTrue(session.isTrashed(file1.getRef()));
+        Boolean isTrashed = session.getDocumentSystemProp(file1.getRef(), SYSPROP_IS_TRASHED, Boolean.class);
         assertTrue(isTrashed != null && isTrashed.booleanValue());
-    }
+        assertTrue(trashService.isMangledName(file1.getName()));
+
+        // follow delete transition to check if TrashService is called
+        trashService.trashDocument(folder);
+
+        txFeature.nextTransaction();
+
+        folder = session.getDocument(folder.getRef());
+        file1 = session.getDocument(file1.getRef());
+        file2 = session.getDocument(file2.getRef());
 
+        assertTrue(session.isTrashed(folder.getRef()));
+        assertTrue(session.isTrashed(file1.getRef()));
+        assertTrue(session.isTrashed(file2.getRef()));
+
+        assertTrue(trashService.isMangledName(folder.getName()));
+        assertFalse(trashService.isMangledName(file2.getName()));
```

Comment:
I have no idea why this assertion is expected, please add a comment to indicate the action and explain briefly what is a mangled name and why it is expected (the JavaDoc doesn't help)

Reply:
OK, I'll also update
>// follow delete transition to check if TrashService is called

as it is now meaningless

Reply:
Did so, but the test is moving in AbstractTestTrashService.
Actually this PR is gonna be clause because this ticket is not for master anymore but only 10.10
[A new ticket has been opened to change the behavior in 11.1](https://jira.nuxeo.com/browse/NXP-28283)